src/core/counter/tables.rs

```diff
@@ -0,0 +1,380 @@
+use crate::core::sizage::Sizage;
+use crate::error::{Error, Result};
+
+pub(crate) fn sizage(s: &str) -> Result<Sizage> {
+    Ok(match s {
+        "-A" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-B" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-C" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-D" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-E" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-F" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-G" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-H" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-I" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-J" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-K" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-L" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-V" => Sizage {
+            hs: 2,
+            ss: 2,
+            fs: 4,
+            ls: 0,
+        },
+        "-0V" => Sizage {
+            hs: 3,
+            ss: 5,
+            fs: 8,
+            ls: 0,
+        },
+        "--AAA" => Sizage {
+            hs: 5,
+            ss: 3,
+            fs: 8,
+            ls: 0,
+        },
+        _ => {
+            return Err(Box::new(Error::UnknownSizage(format!(
+                "{{\
+                    \"message\":\"unknown sizage\",\
+                    \"code\":{s}\
+                }}"
+            ))));
+        }
+    })
+}
+
+pub(crate) fn hardage(s: &str) -> Result<u32> {
+    match s {
+        "-A" | "-B" | "-C" | "-D" | "-E" | "-F" | "-G" | "-H" | "-I" | "-J" | "-K" | "-L"
+        | "-V" => Ok(2),
+        "-0" => Ok(3),
+        "--" => Ok(5),
+        _ => Err(Box::new(Error::UnknownHardage(format!(
+            "{{\
+                \"message\":\"unknown hardage\",\
+                \"code\":{s}\
+            }}"
+        )))),
+    }
+}
+
+pub(crate) fn bardage(b: &[u8]) -> Result<u32> {
```

Comment:
We should think about computing these for other hardages.

Reply:
yeah that's a great suggestion. Can you open an issue?